src/compilers.rkt

```diff
@@ -11,9 +14,68 @@
       [supported supported-list?])]
     [compilers (parameter/c (listof compiler?))]))
 
+;; Compiler struct
+
 (struct compiler (extensions header export footer supported))
 
 (define compilers (make-parameter '()))
 
 (define-syntax-rule (define-compiler arg ...)
   (compilers (cons (compiler arg ...) (compilers))))
+
+;;; Compiler contexts
+
+(define *used-names* (make-parameter (mutable-set)))  ; todo - make immutable?
+(define *gensym-divider* (make-parameter #\_))
+(define *gensym-collisions* (make-parameter 1))
+(define *gensym-unique* (make-parameter #t))
+
+; Returns a unique, printable name based on the symbol and a fix-name procedure.
+(define (gensym sym fix-name) 
+  (define name
+    (if (*gensym-unique*)
+        (for/fold ([gen (fix-name (if (symbol? sym) (symbol->string sym) sym))])
+                  ([i (in-naturals (*gensym-collisions*))]                      
+                  #:break (not (set-member? (*used-names*) gen)))              
+            (*gensym-collisions* (add1 i))                                  
+            (fix-name (format "~a~a~a" sym (*gensym-divider*) i)))
+        (symbol->string sym)))
```

Comment:
This feature is a bit more complicated: you have to avoid collisions with already generated, random symbols. So you would have to check where the symbol is added to the name map.

Since this feature isn't particularly important, I would remove it for now